locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/JcloudsLocation.java

```diff
@@ -2133,11 +2133,17 @@ protected NodeMetadata findNodeOrThrow(ConfigBag config) {
         }
         NodeMetadata node = Iterables.getOnlyElement(candidateNodes);
 
-        String pkd = LocationConfigUtils.getOsCredential(config).checkNoErrors().logAnyWarnings().getPrivateKeyData();
+        OsCredential osCredentials = LocationConfigUtils.getOsCredential(config).checkNoErrors().logAnyWarnings();
+        String pkd = osCredentials.getPrivateKeyData();
+        String password = osCredentials.getPassword();
         if (Strings.isNonBlank(pkd)) {
             LoginCredentials expectedCredentials = LoginCredentials.fromCredentials(new Credentials(user, pkd));
             //override credentials
             node = NodeMetadataBuilder.fromNodeMetadata(node).credentials(expectedCredentials).build();
+        } else if (Strings.isNonBlank(password)) {
+            LoginCredentials expectedCredentials = LoginCredentials.fromCredentials(new Credentials(user, password));
+            //override credentials
+            node = NodeMetadataBuilder.fromNodeMetadata(node).credentials(expectedCredentials).build();
         }
 
         return node;
```

Comment:
@aledsage it was a minor suggestion. I guess, we can just return the node:
```
return NodeMetadataBuilder.fromNodeMetadata(node).credentials(expectedCredentials).build();
```
and remove lines 2145-2146 and 2141-2142, maybe.
if we pulled it out from the if-else and you can